### PR TITLE
Added .sample() for arrays

### PR DESCRIPTION
**Pull request by quintin-reineccius**
Thought this might be useful. Working and Tested 

If you have an array 

``` js
var array = [1, 2, 3, 4, 5, 6, 7, 8, 9]
```

Then you can use .sample() to randomly select a value

``` js
array.sample() -> random value
//like
array.sample() -> 7
```

You can also declare how many values you want and they get returned in an array 

``` js
array.sample(4) -> [random value, random value, random value, random value]
//like
array.sample(4) -> [4, 2, 8, 5]
```

### CODE REVIEW

File: tests/array/sample.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+var chai = require('chai');
+var expect = chai.expect;
+
+require('../../array/sample');
+
+describe('sample', function() {
+  it('should grab random value from array and return it', function() {
+    var array = [1,2,3,4,5,6,7,8,9];
+
+    expect(typeof array.sample()).to.eql("number");
```
Comment (dphaener) [inline tests/array/sample.js]: Hmmmm. How about you test that the returned array has only one element, and that the element was in the original array.

File: tests/array/sample.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  });
+
+  it('should grab random values from array and return an array', function() {
+    var array = [1,2,3,4,5,6,7,8,9];
+
+    expect(typeof array.sample(4)).to.eql("object");
+    expect(array.sample(4).length).to.eql(4);
```
Comment (dphaener) [inline tests/array/sample.js:17]: Same here. Make sure the sampled numbers were all in the original array.

**Comment by quintin-reineccius** [comment]
@dphaener is that ok to use a method we made in another test?

File: array/sample.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    value: function(num){
+      var clone = this.slice(0);
+      var returnedArray = [];
+      var n = num ? num : 1;
+
+      if(n == 1){
+        return clone[Math.floor(Math.random()*clone.length)];
+      }
+      else{
+        for(var i = 0; i < n; i++){
```
Comment (zackify) [inline array/sample.js]: You should try doing .map here

Comment (zackify) [inline array/sample.js]: Eh now that I think about it I'll just wait and see what Darin says

File: array/sample.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+if (!Array.prototype.sample) {
+  Object.defineProperty(Array.prototype, "sample", {
+    value: function(num){
+      var clone = this.slice(0);
+      var returnedArray = [];
+      var n = num ? num : 1;
+
+      if(n == 1){
+        return clone[Math.floor(Math.random()*clone.length)];
```
Comment (zackify) [inline array/sample.js:9]: There's no need to have an else statement here. If n == 1, the function will be exited.

**Pull request merged**